src/prefect/engine/task_runner.py

```diff
@@ -666,6 +672,16 @@ def check_target(self, state: State, inputs: Dict[str, Result]) -> State:
         target = self.task.target
 
         if result and target:
+            if not isinstance(target, str):
+                # Pass all necessary kwargs to target callable
+                raw_inputs = {k: r.value for k, r in inputs.items()}
+                formatting_kwargs = {
+                    **prefect.context.get("parameters", {}).copy(),
+                    **raw_inputs,
+                    **prefect.context,
+                }
+                target = target(**formatting_kwargs)
+
```

Comment:
I think you should create these `formatting_kwargs` in line 675 above and use them here + in lines 685 and 686 for consistency.